vmm/src/device_manager.rs

```diff
@@ -383,8 +392,25 @@ impl DeviceManager {
                 .map_err(DeviceManagerError::BusError)?;
         }
 
-        let mut virtio_devices: Vec<Box<dyn vm_virtio::VirtioDevice>> = Vec::new();
+        let mut virtio_devices: Vec<(Box<dyn vm_virtio::VirtioDevice>, bool)> = Vec::new();
+
+        // Create virtio-iommu as one of the first device, and particularly
+        // before the creation of other virtio and VFIO devices. This is
+        // required since it needs to be provided to the other devices as a
+        // trait that implements some address translations.
+        let _iommu_mapping: Option<Arc<IommuMapping>> = if vm_info.vm_cfg.iommu {
+            let (iommu_device, mapping) =
+                vm_virtio::Iommu::new().map_err(DeviceManagerError::CreateVirtioIommu)?;
+            virtio_devices.push((
+                Box::new(iommu_device) as Box<dyn vm_virtio::VirtioDevice>,
+                false,
+            ));
+            Some(mapping)
+        } else {
+            None
+        };
```

Comment:
Could you move this down into the `pci_support` block below? Then you could get away without naming the variable with the _ prefix.

Reply:
Yes good point!